pkg/webhook/resources/storageclass/validator.go

```diff
@@ -196,25 +203,34 @@ func (v *storageClassValidator) validateEncryption(newObj runtime.Object) error
 
 func (v *storageClassValidator) validateEncryptionSecret(secret *corev1.Secret, secretNamespace string, secretName string) error {
 	invalid := false
-	requiredFields := map[string]string{
-		lhtypes.CryptoKeyCipher:   lhcrypto.CryptoKeyDefaultCipher,
-		lhtypes.CryptoKeyHash:     lhcrypto.CryptoKeyDefaultHash,
-		lhtypes.CryptoKeySize:     lhcrypto.CryptoKeyDefaultSize,
-		lhtypes.CryptoPBKDF:       lhcrypto.CryptoDefaultPBKDF,
-		lhtypes.CryptoKeyProvider: "secret",
-		lhtypes.CryptoKeyValue:    "",
+	requiredFields := map[string][]string{
+		lhtypes.CryptoKeyCipher:   availableCiphers,
+		lhtypes.CryptoKeyHash:     availableHash,
+		lhtypes.CryptoKeySize:     availableKeySizes,
+		lhtypes.CryptoPBKDF:       availablePBKDFs,
+		lhtypes.CryptoKeyProvider: {"secret"},
+		lhtypes.CryptoKeyValue:    {""},
 	}
 
-	for field, defaultValue := range requiredFields {
+	for field, availableValue := range requiredFields {
 		value, ok := secret.Data[field]
 		if !ok {
 			return werror.NewInvalidError(fmt.Sprintf("secret %s/%s is not a valid encryption secret, missing field: %s", secretNamespace, secretName, field), "")
 		}
 
-		if field == lhtypes.CryptoKeyValue {
-			invalid = string(value) == ""
-		} else {
-			invalid = string(value) != defaultValue
+		switch field {
+		case lhtypes.CryptoKeyProvider:
```

Comment:
@Yu-Jack as you mention at the end of the test plan:
> Although we open different options for encryption, there might be some error messages which node might not support specific encryption like this.

The check here literally validates every possible value in each field but doesn't check if the combination from all fields works. The user can only know if the encryption works after doing it. Is there a document to describe all possible combinations?

Reply:
Unfortunately, we don't know the possible combination, but I created a [doc PR](https://github.com/harvester/docs/pull/682) to tell users that it eventually depends on the nodes they're using.